predict.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""torchgeo model inference script."""
+
+import argparse
+import os
+from typing import Dict, Tuple, Type, cast
+
+import pytorch_lightning as pl
+import rasterio as rio
+import torch
+from kornia.contrib import CombineTensorPatches
+from omegaconf import OmegaConf
+
+from torchgeo.datamodules import (
+    BigEarthNetDataModule,
+    ChesapeakeCVPRDataModule,
+    COWCCountingDataModule,
+    CycloneDataModule,
+    ETCI2021DataModule,
+    EuroSATDataModule,
+    InriaAerialImageLabelingDataModule,
+    LandCoverAIDataModule,
+    NAIPChesapeakeDataModule,
+    OSCDDataModule,
+    RESISC45DataModule,
+    SEN12MSDataModule,
+    So2SatDataModule,
+    UCMercedDataModule,
+)
+from torchgeo.trainers import (
+    BYOLTask,
+    ClassificationTask,
+    MultiLabelClassificationTask,
+    RegressionTask,
+    SemanticSegmentationTask,
+)
+
+TASK_TO_MODULES_MAPPING: Dict[
+    str, Tuple[Type[pl.LightningModule], Type[pl.LightningDataModule]]
+] = {
+    "bigearthnet": (MultiLabelClassificationTask, BigEarthNetDataModule),
+    "byol": (BYOLTask, ChesapeakeCVPRDataModule),
+    "chesapeake_cvpr": (SemanticSegmentationTask, ChesapeakeCVPRDataModule),
+    "cowc_counting": (RegressionTask, COWCCountingDataModule),
+    "cyclone": (RegressionTask, CycloneDataModule),
+    "eurosat": (ClassificationTask, EuroSATDataModule),
+    "etci2021": (SemanticSegmentationTask, ETCI2021DataModule),
+    "inria": (SemanticSegmentationTask, InriaAerialImageLabelingDataModule),
+    "landcoverai": (SemanticSegmentationTask, LandCoverAIDataModule),
+    "naipchesapeake": (SemanticSegmentationTask, NAIPChesapeakeDataModule),
+    "oscd": (SemanticSegmentationTask, OSCDDataModule),
+    "resisc45": (ClassificationTask, RESISC45DataModule),
+    "sen12ms": (SemanticSegmentationTask, SEN12MSDataModule),
+    "so2sat": (ClassificationTask, So2SatDataModule),
+    "ucmerced": (ClassificationTask, UCMercedDataModule),
+}
+
+
+def write_mask(mask: torch.Tensor, output_dir: str, input_filename: str) -> None:
+    """Write mask to specified output directory."""
+    output_path = os.path.join(output_dir, os.path.basename(input_filename))
+    with rio.open(input_filename) as src:
+        profile = src.profile
+    profile["count"] = 1
+    profile["dtype"] = "uint8"
+    mask = mask.cpu().numpy()
+    with rio.open(output_path, "w", **profile) as ds:
+        ds.write(mask)
+
+
+def main(config_dir: str, predict_on: str, output_dir: str, device: str) -> None:
+    """Main inference loop."""
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Load checkpoint and config
+    conf = OmegaConf.load(os.path.join(config_dir, "experiment_config.yaml"))
+    ckpt = os.path.join(config_dir, "last.ckpt")
```

Comment:
Should these filenames be parameters?

Reply:
The ckpt filename sure. Think the config filename ("experiment_config.yaml") is hard coded.